caddyhttp/push/link_parser.go

```diff
@@ -0,0 +1,70 @@
+package push
+
+import (
+	"regexp"
+	"strings"
+)
+
+const (
+	commaSeparator     = ","
+	semicolonSeparator = ";"
+	equalSeparator     = "="
+)
+
+var (
+	resourceRegexp = regexp.MustCompile("^<(.*?)>;?(.*)$")
+)
+
+type linkResource struct {
+	uri    string
+	params map[string]string
+}
+
+// parseLinkHeader is responsible for parsing Link header and returning list of found resources.
+//
+// Accepted formats are:
+// Link: </resource>; as=script
+// Link: </resource>; as=script,</resource2>; as=style
+// Link: </resource>;</resource2>
+func parseLinkHeader(header string) []linkResource {
+
+	if header == "" {
+		return make([]linkResource, 0)
+	}
+
+	resources := make([]linkResource, 0)
+
+	for _, link := range strings.Split(header, commaSeparator) {
+		l := linkResource{params: make(map[string]string)}
+
+		groups := resourceRegexp.FindAllStringSubmatch(link, -1)
+
+		if len(groups) == 0 {
+			continue
+		}
+
+		l.uri = strings.TrimSpace(groups[0][1])
+
+		for _, param := range strings.Split(strings.TrimSpace(groups[0][2]), semicolonSeparator) {
```

Comment:
Similar potential problem here; possible out of bounds.

Reply:
I've got rid of regexp.